### DISCUSSION TITLE
使用keil编译时，仅能识别*:\Keil\C51\INC目录下的头文件，其子文件夹目录下的头文件无法识别

### DISCUSSION THREAD

**Discussion by zhaoyu-ma** [github0null/eide#356] [Q&A]
你好，
初学小白，使用EIDE时通过Keil编译器进行编译，发现编译时候仅能识别Keil\C51\INC目录下的头文件，其子文件夹目录下的头文件无法识别，将全部头文件放在src文件夹下可以识别。    尝试了很多方法修改includepath 没有效果。
请问是不是因为Built-in implement C/C++ IntelliSense Provider for ms-vscode.cpptools, Not Need to configurate c_cpp_properties.json file.  这个功能导致的无法修改头文件路径呢？

希望能够解惑，非常感谢

**Comment by github0null** [comment]
`#include` 的时候，要写目录名，比如 `#include "STC/xxx.h"`

**或者**

你把你要的目录添加到项目就行了 https://em-ide.com/docs/modules/project_attr

这样你就可以用 `#include "xxx.h"`
